External/Notepad/Storage.swift

```diff
@@ -143,31 +150,36 @@ class Storage: NSTextStorage {
         let indexRange = string.lineRange(for: nsRange)
         let extendedRange = NSUnionRange(editedRange, NSRange(indexRange, in: string))
 
-        applyStyles(extendedRange)
-        super.processEditing()
+        perserveRealEditedRange {
+            applyStyles(extendedRange)
+            super.processEditing()
+        }
```

Comment:
I'm not a fan of calling `super` from within a closure this way.  It works but feels wrong, ya know? 

Related to my earlier comment about loosing the boolean flag, it might work to call `applyStyles(extendedRange)` from within the block, move the `super` call to after the block, and follow up the super call by always setting `overrideSelectionRange` to `nil`.   I think that would encapsulate all the relevant logic nicely, and keep the `super` call out of the block scope. 
Thoughts?

Reply:
We can loose the closure mechanism altogether! Detail is: both the flag + range must be set **before** `super.processEditing()`, and cleaned up afterwards.

Would it be okay with you, perhaps, if we remove the closure mechanism, and merge both things in the same spot? (only reason why I've opted for a closure is the "old behavior preservation" we've got, the code ends up much cleaner).

WDYT? (Thanks!!!!)

Reply:
Would love to lose the closure! Let's try what you described :)